### PR TITLE
Implement Pipelines

### PR DESCRIPTION
**Pull request by aratz-lasa**
This PR introduces Terraform configurations for defining pipelines using the Warpstream provider. It includes tests and an example to illustrate the usage of these pipelines.

**Example Configuration:**
```yaml
terraform {
  required_providers {
    warpstream = {
      source = "warpstreamlabs/warpstream"
    }
  }
}

provider "warpstream" {
  token    = "aks_xxx"
}

resource "warpstream_virtual_cluster" "tf_example_pipelines" {
  name = "vcn_tf_example_pipelines"
}

resource "warpstream_pipeline" "example_pipeline" {
  virtual_cluster_id             = warpstream_virtual_cluster.tf_example_pipelines.id
  name                           = "example_pipeline"
  state                          = "paused"
  configuration_yaml = <<EOT
  input:
    kafka_franz:
        seed_brokers: ["localhost:9092"]
        topics: ["test_topic"]
        consumer_group: "test_topic_cap"

    processors:
        - mapping: "root = content().capitalize()"

  output:
      kafka_franz:
          seed_brokers: ["localhost:9092"]
          topic: "test_topic_capitalized"
  EOT
}

```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Absolutely! Here's a .gitignore file suitable for a Terraform provider repository:
```
Comment (epot) [inline .gitignore:1]: do you really want to put this? :)

File: internal/provider/pipeline_resource.go [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+			},
+			"state": schema.StringAttribute{
+				Description: "The desired operational state of the pipeline. Valid values are 'running' or 'paused'.",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("running", "paused"),
+				},
+			},
+			"configuration_yaml": schema.StringAttribute{
+				Description: "The YAML content defining the input sources, processing steps, and output destinations for the pipeline. This represents the complete configuration for this specific version.",
```
Comment (epot) [inline internal/provider/pipeline_resource.go:102]: should we put a link to our doc here?

**Review by epot — APPROVED** [review]

File: internal/provider/pipeline_resource.go [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+			"state": schema.StringAttribute{
+				Description: "The desired operational state of the pipeline. Valid values are 'running' or 'paused'.",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("running", "paused"),
+				},
+			},
+			"configuration_yaml": schema.StringAttribute{
+				Description: "The YAML content defining the input sources, processing steps, and output destinations for the pipeline. " +
+					"This represents the complete configuration for this specific version. To understand how to set your configuration take a look at: https://docs.warpstream.com/warpstream/configuration/benthos#getting-started",
```
Comment (7imon7ays) [inline internal/provider/pipeline_resource.go:103]: I opened a gitbook diff to rename this slug configuration/bento. We might want to merge that and update this line before merging this pr.

https://app.gitbook.com/o/xuJtdDASiqpJSjc2Fs19/s/jB7FxO8ty4EXO4HsQP4E/~/changes/408

File: internal/provider/pipeline_resource_test.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+			seed_brokers: ["localhost:9092"]
+			topic: "test_topic_capitalized"
+	EOT
+}`
+}
+
+func testPipelineCheck() resource.TestCheckFunc {
+	return resource.ComposeAggregateTestCheckFunc(
+		resource.TestCheckResourceAttrSet("warpstream_pipeline.test_pipeline", "id"),
+		resource.TestCheckResourceAttrSet("warpstream_pipeline.test_pipeline", "type"),
```
Comment (7imon7ays) [inline internal/provider/pipeline_resource_test.go:52]: Is this always going to be `BENTHOS`? If so we could assert that here.

File: internal/provider/utils/yaml_type.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+		return diags
+	}
+	return diags
+}
+
+func StringToYaml(value string) YamlValue {
+	return YamlValue{
+		StringValue: basetypes.NewStringValue(value),
+	}
+}
```
Comment (7imon7ays) [inline internal/provider/utils/yaml_type.go:96]: Well done figuring all this out 💪

**Review by 7imon7ays — APPROVED** [review]

File: internal/provider/pipeline_resource.go [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+	}
+	plan.ID = types.StringValue(c.PipelineID)
+
+	cc, err := r.client.CreatePipelineConfiguration(ctx, api.HTTPCreatePipelineConfigurationRequest{
+		VirtualClusterID:  plan.VirtualClusterID.ValueString(),
+		PipelineID:        c.PipelineID,
+		ConfigurationYAML: plan.ConfigurationYAML.ValueString(),
+	})
+	if err != nil {
+		r.client.DeletePipeline(ctx, api.HTTPDeletePipelineRequest{
```
Comment (7imon7ays) [inline internal/provider/pipeline_resource.go:147]: Looks like here and below the linter isn't happy

File: internal/provider/pipeline_resource.go (hunk 3 above)
Comment (7imon7ays) [reply internal/provider/pipeline_resource.go:103]: Actually I closed that diff and opened this pr which is more comprehensive https://github.com/warpstreamlabs/docs/pull/72

**Pull request merged**